#!/usr/bin/env python2
from dynamic_reconfigure.parameter_generator_catkin import *
PACKAGE = "asr_recognizer_prediction_ism"


gen = ParameterGenerator()

size_enum = gen.enum([ gen.const("best_path",      int_t, 0, "Best Path"),
                       gen.const("old_prediction_non_normalized", int_t, 1, "old_prediction_non_normalized"),
                       gen.const("old_prediction_normalized", int_t, 2, "old_prediction_normalized"),
                       gen.const("random_path", int_t, 3, "random_path"),
                       gen.const("shortest_path", int_t, 4, "shortest_path")],
                       "An enum to set size")

gen.add("posePredictor", int_t, 0, "Choose Posepredictor.", 4, 0, 4, edit_method=size_enum)
gen.add("enableVisualization", bool_t, 0, "toggle Visualization.", True)


exit(gen.generate(PACKAGE, "asr_recognizer_prediction_ism", "pose_prediction"))
